mlflow/utils/__init__.py

```diff
@@ -203,6 +204,27 @@ def find_free_port():
         return s.getsockname()[1]
 
 
+def check_port_connectivity():
+    port = find_free_port()
+    try:
+        with subprocess.Popen(
+            ["nc", "-l", "-p", str(port)],
```

Comment:
Is `nc` available on Windows systems?

Reply:
No, can we perform this check only on Databricks?

Reply:
like:

```
should_use_stdin_server = is_in_databricks() && !check_port_connectivity()

if should_use_stdin_server:
    ...
```